hack/test-e2e-local.sh

```diff
@@ -19,6 +19,16 @@ ginkgo_flags=
 if [ -n "${CI:-}" -a -n "${ARTIFACTS:-}" ] ; then
   mkdir -p "$ARTIFACTS"
   ginkgo_flags="--output-dir=$ARTIFACTS --junit-report=junit.xml"
+  
+  printf "\n127.0.0.1 api.default.local.external.local.gardener.cloud\n127.0.0.1 api.default.local.internal.local.gardener.cloud" >> /etc/hosts
+else
+cat <<EOF | sudo tee -a /etc/hosts
+
+# Manually created to access the test Gardener shoot clusters with name 'default' in the 'garden-local' namespace.
+# TODO: Remove this again when the shoot cluster access is no longer required.
+127.0.0.1 api.default.local.external.local.gardener.cloud
+127.0.0.1 api.default.local.internal.local.gardener.cloud
+EOF
```

Comment:
Wouldn't this always add them to `/etc/hosts` even if they are already present, eventually polluting the file? While this makes sense for the CI case (new pod always, no persisted `/etc/hosts` file), I think we should rely on the user to properly prepare the file before running the test.

In https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md#creating-a-shoot-cluster, we already state:

> [...] Hence, if you want to access the shoot cluster, you have to run the following command which will extend your /etc/hosts file with the required information to make the DNS names resolvable: [...]

Also, we recommend to add 

```
127.0.0.1 api.e2e-local.local.external.local.gardener.cloud
127.0.0.1 api.e2e-local.local.internal.local.gardener.cloud
```

Does this still apply now? Probably it should be renamed? Maybe the `Shoot` should be called `e2e-default`?

Reply:
> I think we should rely on the user to properly prepare the file before running the test.

So print these instructions and warn (that the test would fail) in the script?

Reply:
You could `grep` for the needed lines in the script and refer to https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md#creating-a-shoot-cluster when they are missing, yeah, maybe something like this

Reply:
> Does this still apply now? 

I'm not sure. @timebertt, WDYT?

Reply:
I think we needed this,
```
127.0.0.1 api.e2e-local.local.external.local.gardener.cloud
127.0.0.1 api.e2e-local.local.internal.local.gardener.cloud
```
to run `make test-e2e-local-fast`, but that just runs the default create_and_delete test.
The shoot isn't named e2e-local in that case too.

Reply:
> Maybe the Shoot should be called e2e-default?

yeah.

Reply:
OK, thanks for checking, sounds like an artefact then which we can replace 👍🏻

Reply:
I think we missed to adapt the docs after this PR.
https://github.com/gardener/gardener/pull/5320/
There was only one test before, and the shoot was called `e2e-local`.
I will make the changes in the doc accordingly.